internal/driver/ipmi/ipmi.go

```diff
@@ -56,21 +57,38 @@ type server struct {
 
 // Cleanly disconnect from the console.
 //
-// This kills the running ipmitool process, and then makes a call to
-// deactivate sol.
+// This injects the shutdown command ".~" into the the impitool process's Stdin,
+// and then after a grace period, kills the process. It also runs ipmitool ...
+// sol deactivate which (imperically) is necessary on some OBMs, but not all.
 func (p *ipmitoolProcess) Shutdown() error {
-	p.proc.Signal(syscall.SIGTERM)
+	_, errWrite := p.conn.Write([]byte("~.\n"))
```

Comment:
The only thing I am confused about is, if this is in fact written to the stdin of the ipmitool process, then we should never have to send SIGKILL or SIGTERM. 

I'll investigate this more later, kinda bummed out due to my lack of understanding of subprocesses generally.

Reply:
In theory you're right, but adding a timeout isn't too hard, and it means if something goes haywire, we throw an error instead of deadlocking.